depthai_nodes/ml/parsers/base_parser.py

```diff
@@ -0,0 +1,33 @@
+from abc import ABCMeta, abstractmethod
+from typing import Dict, Union
+
+import depthai as dai
+
+
+class BaseMeta(ABCMeta, type(dai.node.ThreadedHostNode)):
+    pass
+
+
+class BaseParser(dai.node.ThreadedHostNode, metaclass=BaseMeta):
+    """Base parser class for neural network output parsers."""
```

Comment:
I think its crucial we add thorough explanation of methods and parameters to the `BaseParser` (they were added in previous commits but are now gone).